### ISSUE TITLE
[CI] S3RepositoryThirdPartyTests testCleanup failing

### ISSUE THREAD

**Issue by slobodanadamovic** [elastic/elasticsearch#92000]
**Build scan:**
https://gradle-enterprise.elastic.co/s/6culsltve2urs/tests/:modules:repository-s3:s3ThirdPartyTest/org.elasticsearch.repositories.s3.S3RepositoryThirdPartyTests/testCleanup

**Reproduction line:**
```
./gradlew ':modules:repository-s3:s3ThirdPartyTest' --tests "org.elasticsearch.repositories.s3.S3RepositoryThirdPartyTests.testCleanup" -Dtests.seed=E60D544539AECD62 -Dtests.locale=ko -Dtests.timezone=Europe/Helsinki -Druntime.java=17
```

**Applicable branches:**
8.5

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.repositories.s3.S3RepositoryThirdPartyTests&tests.test=testCleanup

**Failure excerpt:**
```
org.elasticsearch.common.util.concurrent.UncategorizedExecutionException: Failed execution

  at __randomizedtesting.SeedInfo.seed([E60D544539AECD62:B62707EB40035DD7]:0)
  at org.elasticsearch.common.util.concurrent.FutureUtils.rethrowExecutionException(FutureUtils.java:80)
  at org.elasticsearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:50)
  at org.elasticsearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:26)
  at org.elasticsearch.repositories.AbstractThirdPartyRepositoryTestCase.deleteAndAssertEmpty(AbstractThirdPartyRepositoryTestCase.java:71)
  at org.elasticsearch.repositories.AbstractThirdPartyRepositoryTestCase.tearDown(AbstractThirdPartyRepositoryTestCase.java:62)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.util.concurrent.ExecutionException: java.io.IOException: Failed to delete blobs [8.5_third_party_tests_E60D544539AECD62/index-2, 8.5_third_party_tests_E60D544539AECD62/index.latest, 8.5_third_party_tests_E60D544539AECD62/]

    at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:257)
    at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:244)
    at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:75)
    at org.elasticsearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:45)
    at org.elasticsearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:26)
    at org.elasticsearch.repositories.AbstractThirdPartyRepositoryTestCase.deleteAndAssertEmpty(AbstractThirdPartyRepositoryTestCase.java:71)
    at org.elasticsearch.repositories.AbstractThirdPartyRepositoryTestCase.tearDown(AbstractThirdPartyRepositoryTestCase.java:62)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:833)

    Caused by: java.io.IOException: Failed to delete blobs [8.5_third_party_tests_E60D544539AECD62/index-2, 8.5_third_party_tests_E60D544539AECD62/index.latest, 8.5_third_party_tests_E60D544539AECD62/]

      at org.elasticsearch.repositories.s3.S3BlobContainer.doDeleteBlobs(S3BlobContainer.java:356)
      at org.elasticsearch.repositories.s3.S3BlobContainer.delete(S3BlobContainer.java:300)
      at org.elasticsearch.repositories.AbstractThirdPartyRepositoryTestCase.lambda$deleteAndAssertEmpty$0(AbstractThirdPartyRepositoryTestCase.java:70)
      at org.elasticsearch.action.ActionRunnable$1.doRun(ActionRunnable.java:34)
      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:892)
      at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
      at java.lang.Thread.run(Thread.java:833)

      Caused by: com.amazonaws.services.s3.model.MultiObjectDeleteException: One or more objects could not be deleted (Service: null; Status Code: 200; Error Code: null; Request ID: T7YX3SFP2TA5GB68; S3 Extended Request ID: oRoMXjf+WaqLAU4WbKzp8EQkaW0GoY2QbX4N/JtEl5HT4yVZB1+fdnGpFKcQJpla1LiuFduPJuw=; Proxy: null)

        at com.amazonaws.services.s3.AmazonS3Client.deleteObjects(AmazonS3Client.java:2345)
        at org.elasticsearch.repositories.s3.S3BlobContainer.deletePartition(S3BlobContainer.java:362)
        at org.elasticsearch.repositories.s3.S3BlobContainer.lambda$doDeleteBlobs$6(S3BlobContainer.java:349)
        at org.elasticsearch.repositories.s3.SocketAccess.lambda$doPrivilegedVoid$0(SocketAccess.java:46)
        at java.security.AccessController.doPrivileged(AccessController.java:318)
        at org.elasticsearch.repositories.s3.SocketAccess.doPrivilegedVoid(SocketAccess.java:45)
        at org.elasticsearch.repositories.s3.S3BlobContainer.doDeleteBlobs(S3BlobContainer.java:340)
        at org.elasticsearch.repositories.s3.S3BlobContainer.delete(S3BlobContainer.java:300)
        at org.elasticsearch.repositories.AbstractThirdPartyRepositoryTestCase.lambda$deleteAndAssertEmpty$0(AbstractThirdPartyRepositoryTestCase.java:70)
        at org.elasticsearch.action.ActionRunnable$1.doRun(ActionRunnable.java:34)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:892)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(Thread.java:833)

```

**Comment by slobodanadamovic** [comment]
Also, the [S3RepositoryThirdPartyTests » testCreateSnapshot](https://gradle-enterprise.elastic.co/s/6culsltve2urs/tests/:modules:repository-s3:s3ThirdPartyTest/org.elasticsearch.repositories.s3.S3RepositoryThirdPartyTests/testCreateSnapshot) also failed in the same build with error: `org.elasticsearch.ResourceAlreadyExistsException: index [test-idx-1/DvBcS4BPQOyzg92JdJ5Rjg] already exists`

I'm assuming it's just because the previously run `testCleanup` failed to cleanup the index?

**Comment by original-brownbear** [comment]
This is an interesting issue. The failure we ran into here seems to just be some standard bulk delete exception that we see with the S3 repo at a certain rate. 
Now if we try to hard assert that deleting all blobs always works, then that's just not going to work out.
Let me see how we can relax that assertion ...

**Comment by arteam** [comment]
I believe this issue has been fixed by introducing bulk deletes in #98948. There have been no failures for this test over the last month.
